test/e2e/container_iface_name_test.go

```diff
@@ -0,0 +1,580 @@
+		// bridge network is ethX when interface_name in the
+		// containers.conf file is set to default value, i.e., "".
+
+		netName1 := createNetworkName("bridge")
+		netName2 := createNetworkName("bridge")
+
+		defer podmanTest.removeNetwork(netName1)
+		nc1 := podmanTest.Podman([]string{"network", "create", netName1})
+		nc1.WaitWithDefaultTimeout()
+		Expect(nc1).Should(ExitCleanly())
+
+		// Inspect the network configuration
+		nic1 := podmanTest.Podman([]string{"network", "inspect", netName1, "--format", "{{.NetworkInterface}}"})
+		nic1.WaitWithDefaultTimeout()
+		Expect(nic1).Should(ExitCleanly())
+
+		// Once a container executes a new network, the nic will be
+		// created. We should clean those up best we can
+		defer deleteNetworkDevice(nic1.OutputToString())
+
+		defer podmanTest.Podman([]string{"rm", "--time=0", "-f", "test"})
+		ctr := podmanTest.Podman([]string{"run", "-dt", "--network", netName1, "--name", "test", ALPINE, "top"})
+		ctr.WaitWithDefaultTimeout()
+		Expect(ctr).Should(ExitCleanly())
+
+		exec1 := podmanTest.Podman([]string{"exec", "test", "ip", "addr", "show", "eth0"})
+		exec1.WaitWithDefaultTimeout()
+		Expect(exec1).Should(ExitCleanly())
+
+		defer podmanTest.removeNetwork(netName2)
+		nc2 := podmanTest.Podman([]string{"network", "create", netName2})
+		nc2.WaitWithDefaultTimeout()
+		Expect(nc2).Should(ExitCleanly())
+
+		// Inspect the network configuration
+		nic2 := podmanTest.Podman([]string{"network", "inspect", netName2, "--format", "{{.NetworkInterface}}"})
+		nic2.WaitWithDefaultTimeout()
+		Expect(nic2).Should(ExitCleanly())
+
+		// Once a container executes a new network, the nic will be
+		// created. We should clean those up best we can
+		defer deleteNetworkDevice(nic2.OutputToString())
+
+		conn := podmanTest.Podman([]string{"network", "connect", netName2, "test"})
+		conn.WaitWithDefaultTimeout()
+		Expect(conn).Should(ExitCleanly())
+		Expect(conn.ErrorToString()).Should(Equal(""))
+
+		exec2 := podmanTest.Podman([]string{"exec", "test", "ip", "addr", "show", "eth1"})
+		exec2.WaitWithDefaultTimeout()
+		Expect(exec2).Should(ExitCleanly())
+
+		rm := podmanTest.Podman([]string{"rm", "--time=0", "-f", "test"})
+		rm.WaitWithDefaultTimeout()
+		Expect(rm).Should(ExitCleanly())
+		Expect(rm.ErrorToString()).To(Equal(""))
+	})
+
+	It("podman container interface name with default scheme for macvlan network with no parent", func() {
+		// Assert that the network interface name inside container for
+		// macvlan network with no parent interface is ethX when
+		// interface_name in the containers.conf file is set to default
+		// value, i.e., "".
+
+		netName1 := createNetworkName("macvlan")
+		netName2 := createNetworkName("macvlan")
+
+		// There is no nic created by the macvlan driver.
+		defer podmanTest.removeNetwork(netName1)
+		nc1 := podmanTest.Podman([]string{"network", "create", "-d", "macvlan", "--subnet", "10.10.0.0/24", netName1})
```

Comment:
All these tests use the same subnets, `10.10.0` and `10.25.40`. All these tests also run in parallel. @containers/podman-maintainers will this cause collisions and flakes if two of these tests happen to run the same network commands at the same time?

Reply:
That shouldn't be an issue.